tests/testsuite/config.rs

```diff
@@ -68,7 +68,7 @@ fn new_config(env: &[(&str, &str)]) -> Config {
     config
 }
 
-fn assert_error(error: CargoError, msgs: &str) {
+fn assert_error(error: &CargoError, msgs: &str) {
```

Comment:
Similarly, would using `AsRef<CargoError>` avoid changing all the existing use sites?

Reply:
Hmm I could not get these and the path2url to work with AsRef, but I can revert the change altogether if you want.

Reply:
I'll take a swing at it.

Reply:
Here's the change that worked for me on `path2url`:

```diff
-pub fn path2url(p: PathBuf) -> Url {
-    Url::from_file_path(&*p).ok().unwrap()
+pub fn path2url<P: AsRef<Path>>(p: P) -> Url {
+    Url::from_file_path(p).ok().unwrap()
 }
```

Reply:
Thanks a lot!